charmcraft/linters.py

```diff
@@ -515,6 +515,57 @@ def run(self, basedir: pathlib.Path) -> str:
         return self.Result.OK
 
 
+class AdditionalFiles(Linter):
+    """Check that the charm does not contain any additional files in the prime directory.
+
+    A few generated files are ignored.
+    """
+
+    name = "additional-files"
+    text = "No additional files found in the charm."
+    url = "https://juju.is/docs/sdk/include-extra-files-in-a-charm"
+
+    INGNORE_FILES: set[pathlib.Path] = {
```

Comment:
Should `const.CHARM_OPTIONAL_FILES` be ignored?

Reply:
I think these files are automatically staged by the charm part? So they always exist in the `stage`. But we can ignore them just in case.

Reply:
It depends. If we change the plugins or something I don't want this being overlooked.